src/views/Export/LeadsSelection/index.tsx

```diff
@@ -238,21 +335,22 @@ function ExportLeadsTable(props: ComponentProps) {
         }, [headers, activeSort, setActiveSort],
     );
 
+    useEffect(() => {
+        setSelectedLeads(leads.filter(v => v.selected).map(v => v.id));
+    }, [setSelectedLeads, leads]);
```

Comment:
We should avoid doing a setState on useEffect when it's avoidable.

Reply:
here I believe it is necessary.

Reply:
Do we not have control when we change leads? we should just do this over there.